### DISCUSSION TITLE
Sensors down after soup upgrade to 2.3.40

### DISCUSSION THREAD

**Discussion by ghost** [Security-Onion-Solutions/securityonion#3647] [Unsupported Versions] [answered]
This morning, I upgraded from 2.3.21 to 2.3.40 and now none of my sensors work. I have 5 sensors and one manager on a distributed system. My local sensor and the manager are both running Centos 7. My remotes are running Ubuntu 18.04. Soup ran fine. Before upgrading, everything in the SOC was online. After upgrading the SOC showed all of my sensors as offline. I logged into each on and ran `so-status`. Every container status was "ERROR". I tried restarting a few of them and then so-status showed them as "MISSING". I rebooted with same results. When I run `salt \* state.highstate` or `so-test` they return errors similar to this:

**_ID: strelka_coordinator
    Function: docker_container.running
        Name: so-strelka-coordinator
      Result: False
     Comment: Failed to pull somanager:5000/security-onion-solutions/so-redis:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manageripaddress:5000: connect: connection refused
     Started: 19:07:30.187033
    Duration: 253.864 ms
     Changes:_**

That's one example of many

I also get this:

**_[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-redis:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manageripaddress:5000: connect: connection refused_**

Nothing has changed on my firewalls. It all worked just a few hours ago. `ss -tulwn` shows that port 5000 is listening.

These are some errors from  soup.log

This:

**_Performing upgrade from Security Onion 2.3.21 to Security Onion 2.3.40.
Updating dockers to 2.3.40.
Error response from daemon: No such container: so-dockerregistry
Error: No such container: so-dockerregistry_**

But then underneath this it shows dockerregistry "is in the correct state"

And this:

**_Error: No such object: quay.io/securityonion/so-acng:2.3.40
jq: error (at <stdin>:1): Cannot iterate over null (null_)**

Everything finished and there were no further errors in that log. After that, my sensors went down.

I also have filebeat, elastalert, kibana and thehive logs that were updated today. 

This is repeated a ton of times in filebeat.log:

**_2021-03-26T17:41:59.015Z	ERROR	[logstash]	logstash/async.go:280	Failed to publish events caused by: EOF
2021-03-26T17:41:59.183Z	ERROR	[logstash]	logstash/async.go:280	Failed to publish events caused by: client is not connected
2021-03-26T17:42:00.612Z	ERROR	[publisher_pipeline_output]	pipeline/output.go:180	failed to publish events: client is not connected_**

### thehive log

**_[2021-03-26T17:46:48,428][ERROR][org.elasticsearch.xpack.monitoring.exporter.local.LocalExporter] failed to set monitoring pipeline [xpack_monitoring_6]
org.elasticsearch.cluster.metadata.ProcessClusterEventTimeoutException: failed to process cluster event (put-pipeline-xpack_monitoring_6) within 30s
	at org.elasticsearch.cluster.service.MasterService$Batcher.lambda$onTimeout$0(MasterService.java:132) [elasticsearch-7.11.2.jar:7.11.2]
	at java.util.ArrayList.forEach(ArrayList.java:1511) [?:?]
	at org.elasticsearch.cluster.service.MasterService$Batcher.lambda$onTimeout$1(MasterService.java:131) [elasticsearch-7.11.2.jar:7.11.2]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673) [elasticsearch-7.11.2.jar:7.11.2]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]_**
	
### kibana log

**_{"type":"log","@timestamp":"2021-03-26T14:02:39Z","tags":["warning","savedobjects-service"],"pid":292,"message":"Unable to connect to Elasticsearch. Error: [search_phase_execution_exception] all shards failed"}

{"type":"log","@timestamp":"2021-03-26T17:47:47+00:00","tags":["fatal","root"],"pid":293,"message":"Error: [.kibana_2][0] primary shard is not active Timeout: [1m], request: [BulkShardRequest [[.kibana_2][0]] containing [100] requests]\n    at Object.write (/usr/share/kibana/src/core/server/saved_objects/migrations/core/elastic_index.js:147:21)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at migrateSourceToDest (/usr/share/kibana/src/core/server/saved_objects/migrations/core/index_migrator.js:216:5)\n    at migrateIndex (/usr/share/kibana/src/core/server/saved_objects/migrations/core/index_migrator.js:127:3)\n    at coordinateMigration (/usr/share/kibana/src/core/server/saved_objects/migrations/core/migration_coordinator.js:53:12)\n    at async Promise.all (index 0)\n    at SavedObjectsService.start (/usr/share/kibana/src/core/server/saved_objects/saved_objects_service.js:153:7)\n    at Server.start (/usr/share/kibana/src/core/server/server.js:283:31)\n    at Root.start (/usr/share/kibana/src/core/server/root/index.js:58:14)\n    at bootstrap (/usr/share/kibana/src/core/server/bootstrap.js:100 : 5)\n    at Command.<anonymous> (/usr/share/kibana/src/cli/serve/serve.js:169 : 5) {\n  detail: {\n    index: {\n      _index: '.kibana_2',\n      _type: '_doc',\n      _id: 'visualization:e017cb80-777b-11ea-bee5-af7f7c7b8e05',\n      status: 503,\n      error: [Object]\n    }\n  }\n}"}

{"type":"log","@timestamp":"2021-03-26T18:51:30+00:00","tags":["warning","plugins","licensing"],"pid":293,"message":"License information could not be obtained from Elasticsearch due to Error: Cluster client cannot be used after it has been closed. error"}_**

I'm not sure what else you need. Any help is appreciated.

Thanks

Troy

**Comment by ghost** [comment]
Can anyone help me with this? I'm completely dead in the water. Not one of my remote sensor nodes will work. 

Ports are open. Everything worked well until the upgrade. I did run  apt update/upgrade on the remote sensors before running soup from the manager. Did I kill everything by updating something I shouldn't have? I tried re-installing but run into these errors.

Thanks.

**_root@soremote-is:~# cat /root/sosetup.log | grep "ERROR"
[ERROR   ] The Salt Master has cached the public key for this node, this salt minion will wait for 10 seconds before attempting to re-authenticate
[ERROR   ] The Salt Master has cached the public key for this node, this salt minion will wait for 10 seconds before attempting to re-authenticate
[ERROR   ] 'mine.send': False
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-nginx:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manager-ip-address:5000: connect: connection refused
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-nginx:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manager-ip-address:5000: connect: connection refused
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-steno:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manager-ip-address:5000: connect: connection refused
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-steno:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manager-ip-address:5000: connect: connection refused
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-suricata:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manager-ip-address:5000: connect: connection refused
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-suricata:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manager-ip-address:5000: connect: connection refused
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-zeek:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manager-ip-address:5000: connect: connection refused
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-zeek:2.3.40: Error 500: Get https://somanager:5000/v2/: dial tcp manager-ip-address:5000: connect: connection refused
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-soc:2.3.40: Unable to perform pull: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.41/images/create?tag=2.3.40&fromImage=somanager%3A5000%2Fsecurity-onion-solutions%2Fso-soc
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-soc:2.3.40: Unable to perform pull: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.41/images/create?tag=2.3.40&fromImage=somanager%3A5000%2Fsecurity-onion-solutions%2Fso-soc
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-telegraf:2.3.40: Unable to perform pull: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.41/images/create?tag=2.3.40&fromImage=somanager%3A5000%2Fsecurity-onion-solutions%2Fso-telegraf
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-telegraf:2.3.40: Unable to perform pull: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.41/images/create?tag=2.3.40&fromImage=somanager%3A5000%2Fsecurity-onion-solutions%2Fso-telegraf
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-nginx:2.3.40: Unable to perform pull: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.41/images/create?tag=2.3.40&fromImage=somanager%3A5000%2Fsecurity-onion-solutions%2Fso-nginx
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-steno:2.3.40: Unable to perform pull: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.41/images/create?tag=2.3.40&fromImage=somanager%3A5000%2Fsecurity-onion-solutions%2Fso-steno
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-suricata:2.3.40: Unable to perform pull: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.41/images/create?tag=2.3.40&fromImage=somanager%3A5000%2Fsecurity-onion-solutions%2Fso-suricata
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-zeek:2.3.40: Unable to perform pull: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.41/images/create?tag=2.3.40&fromImage=somanager%3A5000%2Fsecurity-onion-solutions%2Fso-zeek
[ERROR   ] Failed to pull somanager:5000/security-onion-solutions/so-wazuh:2.3.40: Unable to perform pull: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.41/images/create?tag=2.3.40&fromImage=somanager%3A5000%2Fsecurity-onion-solutions%2Fso-wazuh_**

**Comment by TOoSmOotH** [comment, marked as answer]
Looks like docker registry is messed up. 

sudo docker stop so-dockerregistry
sudo docker rm so-dockerregistry
sudo salt-call state.highstate

Then run a highstate on a minion and see if it connects.

**Reply by ghost** [reply]
Thank you! That worked for three of the five so far. One was shut down by the remote user. I'll get that one tomorrow. One seems to not be able to communicate with the manager at all. When I try to SSH I keep getting 
**_ssh_exchange_identification: Connection closed by remote host_**. I can nmap to the manager from it and it shows as up and lists all of the open ports. I can SSH from the manager to it. I checked the firewall on the manager and this host is listed. However, when I run salt-key, it is not in any of the lists. I looked at the setup log, because I tried to re-run the setup and this is what it shows:

**_Generating SSH key
/root/.ssh/so.key already exists.
Overwrite (y/n)? Removing old entry for manager from known_hosts if it exists
Copying the SSH key to the manager
/usr/bin/ssh-copy-id: INFO: Source of key(s) to be installed: "/root/.ssh/so.key.pub"
ssh_exchange_identification: Connection closed by remote host
Could not determine version of Security Onion running on manager somanager. Please check your network settings and run setup again.
2021-03-29T21:15:29Z | I | Installer version mismatch, downloading correct version from manager
ssh_exchange_identification: Connection closed by remote host
ssh_exchange_identification: Connection closed by remote host
Could not download .tar.gz from manager, please check your network settings and verify the file /opt/so/repo/.tar.gz exists on the manager._**

One other odd thing is that two of the nodes that were successful with the highstate command are showing up in the SOC grid but their IP addresses are no longer listed. Not critical; just odd. 

Thanks again. I appreciate your help!

**Comment by ghost** [comment]
Thank you! I executed all three on the manager. I'm waiting for sudo salt-call state.highstate to finish. I will update when it's done and I've run sudo salt-call state.highstate from the minions.
